### ISSUE TITLE
[rllib] RLlib does not respect Box low/high

### ISSUE THREAD

**Issue by eugenevinitsky** [ray-project/ray#1862]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac OS
- **Ray installed from (source or binary)**: Source
- **Ray version**: 0.4.0
- **Python version**: 3.6
- **Exact command to reproduce**: Any example with a Box

### Describe the problem
If you set the action space to be Box(low=0, high=5) for any Gym env and print out the actions, they actually wind up not being in the desired bounds. It seems like the actions should be clipped?

**Comment by ericl** [comment]
Hmm, we could squash using tanh automatically. @pcmoritz any thoughts?

**Comment by bbalaji-ucsd** [comment]
I'm having the same issue. The action space exploration by any of the agents is approximately [-2, 2] regardless of the Box low, high specification.

**Comment by ericl** [comment]
@bbalaji-ucsd maybe try https://github.com/ray-project/ray/pull/2239/files

I haven't tested how this affects performance though.

**Comment by bbalaji-ucsd** [comment]
Thanks! That solved the problem. Is there any testing I can do to merge this into the main branch?

**Comment by ericl** [comment]
Any before / after benchmark would be helpful. I'm seeing some potential instability after this change on Pendulum / PPO, but that might just be a matter of tuning hyperparams.

**Comment by bbalaji-ucsd** [comment]
I tried running Pendulum / PPO with the changes from the above pull request. After a few iterations, the actions and related metrics like kl-divergence become "nan" and it just loops like that forever. Any idea why that might be? I'll try out other continuous spaces algorithms as well to see what happens. Also, is there any guideline on how to tune the hyper-parameters?

**Comment by ericl** [comment]
Hm, when I print out the actions in `compute_actions()` of loss.py, sometimes they seem to be saturated at

```[[2.]]
[[2.]]
[[2.]]
[[-2.]]
[[2.]]
```

This must mean the network output has grown to such a large magnitude that the output starts drifting randomly. Eventually the action starts becoming `nan` too. Not sure the best way to prevent this, however setting `model: {free_log_std: True}` might make sense.

**Comment by ericl** [comment]
I tried moving the sigmoid to apply to the DiagGaussian means only, and it seems to stabilize the training. I also made squash_to_range a model option in the PR, so @bbalaji-ucsd if the new changes work for you we can go ahead and merge to master.

Btw it seems a beta distribution might be a more principled way of doing this: http://proceedings.mlr.press/v70/chou17a/chou17a.pdf

**Comment by ericl** [comment]
Merged.

**Comment by bbalaji-ucsd** [comment]
Thanks! Sorry didn't get a chance to test it myself. I was looking at RLlab's implementation, and it looks like they simply clip the action after linear scaling. https://github.com/rll/rllab/blob/master/rllab/envs/normalized_env.py#L83 

I'll update here if I find anything new or if your solution works fine.
